include/ginkgo/core/base/lin_op.hpp

```diff
@@ -241,6 +241,22 @@ class LinOp : public EnableAbstractPolymorphicObject<LinOp> {
      */
     virtual bool apply_uses_initial_guess() const { return false; }
 
+    LinOp &operator=(const LinOp &) = default;
+
+    LinOp &operator=(LinOp &&other)
+    {
```

Comment:
I guess we still preserve the executor ?

Reply:
yes, that's an invariant of PolymorphicObject. I should probably call its move-assignment operator here. Great catch!
```suggestion
    {
        EnableAbstractPolymorphicObject<LinOp>::operator=(std::move(other));
```